check/shellcheck

```diff
@@ -30,11 +30,11 @@ for arg in "$@"; do
 done
 
 # Find all shell scripts in this repository.
-IFS=$'\n' read -r -d '' -a our_shell_scripts < \
-    <(git ls-files -z -- \
+IFS=$'\n' read -r -d '' -a our_shell_scripts <<< \
+    "$(git ls-files -z -- \
         ':(exclude)*.'{py,json,json_inward,repr,repr_inward,ipynb,txt,md} \
         ':(exclude)*.'{yaml,ts,tst,rst,pyi,cfg} | \
-        xargs -0 file | grep -i 'shell script' | cut -d: -f1)
+        xargs -0 file | grep -i 'shell script' | cut -d: -f1)"
```

Comment:
If we are mucking with this, can you try this version with mapfile?
I guess the problem could be because of linebreak between `<` and `<(...)` which is on one line below.
`mapfile` is also a cleaner method than overriding IFS.

```shell
mapfile -t our_shell_scripts < <(
    git ls-files -z -- \
        ':(exclude)*.'{py,json,json_inward,repr,repr_inward,ipynb,txt,md} \
        ':(exclude)*.'{yaml,ts,tst,rst,pyi,cfg} |
        xargs -0 file | grep -i 'shell script' | cut -d: -f1
)
```

Reply:
One problem is that `mapfile` is not available in the old version of Bash (3.2) that ships with MacOS, at least in MacOS 15.3. That's not a problem for the GitHub linux runners, but it seems likely that the Mac runners use the stock version of Bash.

Reply:
Under the assumption that this precludes using `mapfile`, I only made the requested formatting change regarding the backslash and not the mapfile change.